src/main/java/org/example/Main.java

```diff
@@ -122,338 +163,205 @@ public boolean isHappy() {
+                    //toSkip |= users.contains(param) && !num.params.contains(param);
+                    //toSkip |= users.contains("-" + param) && num.params.contains(param);
                 }
-            } else {
-                long startNumber = Long.parseLong(result[0]); // is it OK to throw an exception if the first or second param is not a number?
-                int count = Integer.parseInt(result[1]);
-                if (count <= 0) {
-                    System.out.println("The second parameter should be a natural number.");
+                if (toSkip) {
                     continue;
                 }
-                Collection<String> users = new ArrayList<>(); // you can just use diamond operator (<>), no need to specify type
-                // not good to do this in a cycle, because it leads to O(N^2) complexity - better to put all in Set. I don't think you need an order here
+            }
 
-                for (int i = 2; i < result.length; i++) {
-                    if (!users.contains(result[i].toUpperCase())) {
-                        users.add(result[i].toUpperCase());
-                    }
-                }
+            num.printShortInfo();
+            count--;
+        }
+    }
 
-                // let's make it a class constant field: private static final PROPERTIES and store them in UPPER case to avoid doing toLowerCase in a loop below
-                // also better to make it a Set<String> and check .contains instead of looping for each user-defined property
-                String[] properties = {"buzz", "duck", "palindromic",
-                        "gapful", "spy", "even", "odd", "sunny", "square", "jumping",
-                        "happy", "sad"};
-                //                 ArrayList<String> incorrect = new ArrayList<String>(); // when you create a collection it's better to use the most generic interface as a type, e.g. List<String> or even Collection<String>
-                ArrayList<String> incorrect = new ArrayList<String>();
-                for (String user: users) {
-                    boolean correct = false;
-                    for (String p : properties) {
-                        if (user.toLowerCase().equals(p)) {
-                            correct = true;
-                            break;
-                        }
-                        if (user.toLowerCase().equals("-" + p)) {
-                            correct = true;
-                            break;
-                        }
-                    }
-                    if (!correct) {
-                        incorrect.add(user);
-                    }
-                }
+    private static boolean hasInvalidProperty(Collection<String> users) {
+        if (users.isEmpty()) {
+            return false;
+        }
+        // TODO how to create these?
+        Collection<String> allCases =  Stream.of(MyNumber.PARAMS.values()).map(MyNumber.PARAMS::name).toList();
```

Comment:
Collection<String> allCases = Arrays.stream(MyNumber.PARAMS.values()).map(MyNumber.PARAMS::name).collect(Collectors.toList());

Reply:
explain the logic step by step